### ISSUE TITLE
to_png matplotlib deprecation warnings

### ISSUE THREAD

**Issue by moorepants** [sympy/sympy#21739]
In Jupyter notebooks I've been seeing these deprecation warnings:

```
/srv/conda/envs/notebook/lib/python3.9/site-packages/IPython/lib/latextools.py:126: MatplotlibDeprecationWarning: 
The to_png function was deprecated in Matplotlib 3.4 and will be removed two minor releases later. Use mathtext.math_to_image instead.
  mt.to_png(f, s, fontsize=12, dpi=dpi, color=color)
/srv/conda/envs/notebook/lib/python3.9/site-packages/IPython/lib/latextools.py:126: MatplotlibDeprecationWarning: 
The to_rgba function was deprecated in Matplotlib 3.4 and will be removed two minor releases later. Use mathtext.math_to_image instead.
  mt.to_png(f, s, fontsize=12, dpi=dpi, color=color)
/srv/conda/envs/notebook/lib/python3.9/site-packages/IPython/lib/latextools.py:126: MatplotlibDeprecationWarning: 
The to_mask function was deprecated in Matplotlib 3.4 and will be removed two minor releases later. Use mathtext.math_to_image instead.
  mt.to_png(f, s, fontsize=12, dpi=dpi, color=color)
/srv/conda/envs/notebook/lib/python3.9/site-packages/IPython/lib/latextools.py:126: MatplotlibDeprecationWarning: 
The MathtextBackendBitmap class was deprecated in Matplotlib 3.4 and will be removed two minor releases later. Use mathtext.math_to_image instead.
  mt.to_png(f, s, fontsize=12, dpi=dpi, color=color)
```

SymPy 1.8 and matplotlib 3.4.2 and IPython 7.25.0

**Comment by michaelaye** [comment]
very annoying. Here's how to filter them for now:

```
from matplotlib import MatplotlibDeprecationWarning
warnings.filterwarnings("ignore", category=MatplotlibDeprecationWarning)
```

**Comment by oscarbenjamin** [comment]
This also shows up in a SymPy console in Spyder

**Comment by oscarbenjamin** [comment]
Actually the issue with Spyder was fixed in https://github.com/ipython/ipython/pull/12889

**Comment by oscarbenjamin** [comment]
I can reproduce the issue in a jupyter notebook by entering:
```
from sympy import *
x = Symbol('x')
init_printing()
x**2
```
Upgrading to ipython 8.0.1 the warning goes away.

I think this is already fixed in ipython.

**Comment by asmeurer** [comment]
It does look like the warnings are coming from IPython. SymPy also has code to convert LaTeX to png using matplotlib, but it shouldn't be called in the notebook unless it is enabled manually with something like `init_printing(use_latex='matplotlib')` (although things like Spyder or the qtconsole may it automatically). I don't see any instances of `to_png` or `MathtextBackendBitmap ` in the SymPy codebase.

**Comment by oscarbenjamin** [comment]
SymPy uses this function from IPython:
https://github.com/sympy/sympy/blob/a2342a1f56ae8aa67c21636d50e9cb8dba203a9c/sympy/interactive/printing.py#L38
The function was (indirectly) changed by https://github.com/ipython/ipython/pull/12889 and no longer calls deprecated matplotlib APIs.

**Comment by oscarbenjamin** [comment]
I think this is fixed by ipython version 8.0.1 so closing. Please comment here if this is not fixed after upgrading ipython to 8.0.1 or newer.
